Add blog post about Volcano plugin for Headlamp (#55834)
kubernetes/website: content/en/blog/_posts/2026/volcano-headlamp-plugin/index.md

---
layout: blog

> title: "Inspecting Volcano Workloads Faster with kubernetes-sigs/headlamp"

Reviewer: nit: I'd title it:
```suggestion
title: "Inspecting Volcano Workloads Faster with Headlamp"
```

Author: Done

---
layout: blog
title: "Inspecting Volcano Workloads Faster with kubernetes-sigs/headlamp"
date: 2026-05-19
draft: true
slug: visual-context-volcano-headlamp-plugin
---

> **Author:** [Mahmoud Magdy](https://github.com/mahmoudmagdy1-1) (independent)

Reviewer: Thanks, but: this goes in the front matter.

Author: Done

This is especially useful when a workload is pending or not progressing as expected. The map can show the Job, its related PodGroup, the Pods created for the workload, and the Queue context around it. Warning and error states also make it easier to spot resources that need attention.

![Volcano resources in the Headlamp map view](volcano-map-view.png)

## Why use this alongside CLI tools

The plugin is not trying to replace `kubectl` or Volcano CLI. Those remain important for automation, scripting, and raw object inspection. What the plugin improves is the interactive troubleshooting experience: discovering related resources more quickly, understanding structured detail pages, and moving from scheduling state to runtime output without switching tools constantly.

## What’s next

This work brings the main Volcano workflow into Headlamp, including Jobs, Queues, PodGroups, and the map view. Future work will build on this foundation with Prometheus integration, richer scheduling insights, and more workflow-oriented visibility across Volcano workloads.

## Try it and share feedback

To try the plugin:

> 1. Install Headlamp.

Reviewer 2: These numbers are all one, and need to be 1,2,3,4,5.

Author: Done

1. Open the Plugin Catalog from the Headlamp UI.
1. Search for Volcano.
1. Install the Volcano plugin.
1. Connect Headlamp to a Kubernetes cluster where Volcano is already installed.

![Volcano plugin in the Headlamp Plugin Catalog](volcano-plugin-catalog.png)

> If you have ideas, feature requests, or bug reports, open an issue in the plugins repository. Feedback from real Volcano users will help shape what comes next.

Reviewer 2: "in the headlamp plugins repository" with a link to the repo?

Author: Done, Thanks for the review

---
layout: blog
title: "Inspecting Volcano Workloads Faster with kubernetes-sigs/headlamp"
draft: true
slug: visual-context-volcano-headlamp-plugin
author: >
  [Mahmoud Magdy](https://github.com/mahmoudmagdy1-1) (independent)
---

[Volcano](https://volcano.sh/) is a cloud native batch scheduler for Kubernetes, built for high-performance computing, AI/ML, and other batch workloads.

[Headlamp](https://headlamp.dev/) is an extensible Kubernetes web UI. With its plugin system, Headlamp can surface APIs and workflows beyond the built-in Kubernetes resources. The Volcano plugin brings core Volcano resources into Headlamp so you can inspect workload state, queue behavior, and gang scheduling details in one place.

> Watch this short walkthrough to see the Volcano plugin in Headlamp:

Reviewer 3: I was not familiar with Volcano before reading this article, and after the two introduction paragraphs I found myself wanting a short real-world motivation: a paragraph showing in which cases I should consider Volcano (or where it could be useful) before diving into the plugin itself. Something like:

```suggestion
Kubernetes was originally designed for long-running services, where applications are expected to start and remain running for extended periods of time. Batch and HPC workloads have different requirements: jobs arrive dynamically, compete for limited resources, and often need multiple workers to start together.

Volcano extends Kubernetes with concepts such as queues, priorities, quotas, and gang scheduling, making it possible to run AI, ML, and HPC workloads efficiently on shared clusters. Instead of treating every Pod independently, Volcano schedules workloads with awareness of the job as a whole and the resources it requires to make progress.

To make these workloads easier to operate and troubleshoot, the Volcano plugin brings their scheduling context directly into Headlamp.

Watch this short walkthrough to see the Volcano plugin in Headlamp:
```

Author: Done, I added a short motivation section before the walkthrough so readers who are new to Volcano have more context.

{{< youtube id="Mqm1EyAa7TY" title="Volcano plugin for Headlamp walkthrough" >}}

## Visual context helps teams understand Volcano jobs, queues, and PodGroups faster

> Working with Volcano often means moving across several related resources while trying to understand a batch workload. You might start with a Job, then look at the related PodGroup, inspect the Pods behind it, check the Queue, and finally return to the Job again. All of that is possible with CLI tools like `kubectl` and `Volcano CLI`, but it can become fragmented very quickly.

Reviewer 3: Per the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#code-style-inline-code), backticks are for inline code and literal command/tool names (such as `kubectl`). "Volcano CLI" is a descriptive product name rather than a command you type, so it should be plain text: "the Volcano CLI" — or `vcctl` if you mean the actual command.

Author: Fixed it for both, Thank you

The Volcano plugin for Headlamp makes that workflow easier by bringing the key resources together in a single UI. Instead of reconstructing relationships manually, you can move directly between Jobs, Queues, PodGroups, Pods, and events from the same interface.

> The plugin focuses on the three Volcano resources you typically start with: Jobs, Queues, and PodGroups. Each one has dedicated list and detail views inside a Volcano section in the Headlamp sidebar.

Reviewer 3: Since readers may meet these resources here for the first time, it would help to briefly introduce what each Volcano entity is for before describing the plugin views — in the same way the Kubernetes docs describe [cluster components](https://kubernetes.io/docs/concepts/overview/components/). For example:

```suggestion
Volcano introduces its own resources on top of the core Kubernetes objects:

Job
: Describes a batch workload as a set of tasks and the Pods they create.

Queue
: Divides cluster capacity between teams or workloads using quotas and priorities.

PodGroup
: Ties a group of Pods together so that the scheduler treats them as a single unit for gang scheduling.

The plugin surfaces all three resource types directly in Headlamp, providing dedicated list and detail views for each of them under a Volcano section in the sidebar.
```

Author: Thanks, I added brief descriptions for Job, Queue, and PodGroup before describing the plugin views.

## Jobs: workload status, actions, and logs

The Job view is the center of the plugin experience. In the list view, you can quickly understand the basics of a workload, including its status, queue, running versus minimum-available values, task count, and age.

> ![Volcano Jobs list in Headlamp](volcano-jobs-list.png)

Reviewer 3: The alt text is helpful. One blog-guidelines suggestion: for illustrations, the Kubernetes blog guidance recommends using the figure shortcode where feasible.
https://kubernetes.io/docs/contribute/blog/guidelines/#illustrations

```suggestion
{{< figure src="volcano-jobs-list.png" alt="Volcano Jobs list in Headlamp" >}}
```

Author: Fixed, Thanks

The detail view goes further by surfacing the information you usually need while debugging a Job: task details, Pod status, related Queue and PodGroup links, conditions, events, and more. Instead of forcing you to jump between several CLI commands, the plugin keeps that context together in a single page.

The Job page also adds supported lifecycle actions for appropriate states, including **Suspend** and **Resume**, so you can act on a Job directly from the UI.

Another useful addition is direct **Job logs** access. You can open logs for Pods created by a Volcano Job without leaving the Job detail page. The logs viewer supports both single-Pod and all-Pods views, along with container selection and common log controls such as line count, previous logs, timestamps, and follow.

> ![Volcano Job logs in Headlamp](volcano-job-logs.png)

Reviewer 3: Same here.
```suggestion
{{< figure src="volcano-job-logs.png" alt="Volcano Job logs in Headlamp" >}}
```

Author: Fixed, Thanks

## Queues: scheduling capacity and resource context

The Queue view provides much more than a small set of top-level fields. It helps you understand how resources are being allocated and constrained by surfacing capacity, allocated resources, deserved and guaranteed resources, reservation details, child queues, and more.

This makes the Queue page much more useful when trying to understand how resources are being shared and limited across queues.

> ![Volcano Queue details in Headlamp](volcano-queue-detail.png)

Reviewer 3: Same here.
```suggestion
{{< figure src="volcano-queue-detail.png" alt="Volcano Queue details in Headlamp" >}}
```

Author: Done, Thanks

## PodGroups: gang scheduling state and blockers

PodGroups are central to understanding gang scheduling in Volcano, and the plugin makes that state easier to inspect. The PodGroup view highlights progress, conditions, minimum resource requirements, and more.

This also gives you a clearer picture of whether a workload is blocked because it has not yet met the scheduling conditions required to run as a group.

> ![Volcano PodGroup details in Headlamp](volcano-podgroup-detail.png)

Reviewer 3: Same here.
```suggestion
{{< figure src="volcano-podgroup-detail.png" alt="Volcano PodGroup details in Headlamp" >}}
```

Author: Done, Thanks

## Map view: jobs, queues, PodGroups, and pods in one place

The map view shows how Volcano resources are connected. Instead of inspecting each resource separately, you can see how Jobs, PodGroups, Queues, and Pods relate to one another.

This is especially useful when a workload is pending or not progressing as expected. The map can show the Job, its related PodGroup, the Pods created for the workload, and the Queue context around it. Warning and error states also make it easier to spot resources that need attention.

> ![Volcano resources in the Headlamp map view](volcano-map-view.png)

Reviewer 3: Same here.
```suggestion
{{< figure src="volcano-map-view.png" alt="Volcano resources in the Headlamp map view" >}}
```

Author: Fixed, Thanks

## Why use this alongside CLI tools

> The plugin is not trying to replace `kubectl` or `Volcano CLI`. Those remain important for automation, scripting, and raw object inspection. What the plugin improves is the interactive troubleshooting experience: discovering related resources more quickly, understanding structured detail pages, and moving from scheduling state to runtime output without switching tools constantly.

Reviewer 3: Same here: plain "the Volcano CLI" (or `vcctl`).

## What’s next

This work brings the main Volcano workflow into Headlamp, including Jobs, Queues, PodGroups, and the map view. Possible future work includes Prometheus integration, richer scheduling insights, and more workflow-oriented visibility across Volcano workloads.

## Try it and share feedback

To try the plugin:

1. Install Headlamp.
2. Open the Plugin Catalog from the Headlamp UI.
3. Search for Volcano.
4. Install the Volcano plugin.
5. Connect Headlamp to a Kubernetes cluster where Volcano is already installed.

> ![Volcano plugin in the Headlamp Plugin Catalog](volcano-plugin-catalog.png)

Reviewer 3: Same here.
```suggestion
{{< figure src="volcano-plugin-catalog.png" alt="Volcano plugin in the Headlamp Plugin Catalog" >}}
```

Author: Fixed, Thanks

---
layout: blog
title: "Inspecting Volcano Workloads Faster with kubernetes-sigs/headlamp"
date: 2026-05-19
draft: true
slug: visual-context-volcano-headlamp-plugin
author: >
  [Mahmoud Magdy](https://github.com/mahmoudmagdy1-1) (independent)
---

[Volcano](https://volcano.sh/) is a cloud native batch scheduler for Kubernetes, built for high-performance computing, AI/ML, and other batch workloads.

[Headlamp](https://headlamp.dev/) is an extensible Kubernetes web UI. With its plugin system, Headlamp can surface APIs and workflows beyond the built-in Kubernetes resources. The Volcano plugin brings core Volcano resources into Headlamp so you can inspect workload state, queue behavior, and gang scheduling details in one place.

Watch this short walkthrough to see the Volcano plugin in Headlamp:

{{< youtube id="Mqm1EyAa7TY" title="Volcano plugin for Headlamp walkthrough" >}}

## Visual context helps teams understand Volcano jobs, queues, and PodGroups faster

> Working with Volcano often means moving across several related resources while trying to understand a batch workload. You might start with a Job, then look at the related PodGroup, inspect the Pods behind it, check the Queue, and finally return to the Job again. All of that is possible with CLI tools like `kubectl` and Volcano CLI, but it can become fragmented very quickly.

Reviewer 4: nit: kubectl is code-formatted but Volcano CLI isn't

```suggestion
Working with Volcano often means moving across several related resources while trying to understand a batch workload. You might start with a Job, then look at the related PodGroup, inspect the Pods behind it, check the Queue, and finally return to the Job again. All of that is possible with CLI tools like `kubectl` and `Volcano CLI`, but it can become fragmented very quickly.
```

The Volcano plugin for Headlamp makes that workflow easier by bringing the key resources together in a single UI. Instead of reconstructing relationships manually, you can move directly between Jobs, Queues, PodGroups, Pods, and events from the same interface.

> ## The Volcano plugin helps teams inspect jobs, queues, and PodGroups in Headlamp

Reviewer 4: The two H2s (L25 and L26) say almost the same thing back-to-back. 

```
## Visual context helps teams understand Volcano jobs, queues, and PodGroups faster
```

```
## The Volcano plugin helps teams inspect jobs, queues, and PodGroups in Headlamp
```

Consider collapsing into a single section ("What the plugin shows you" or similar)

![Volcano PodGroup details in Headlamp](volcano-podgroup-detail.png)

## Map view: jobs, queues, PodGroups, and pods in one place

The map view shows how Volcano resources are connected. Instead of inspecting each resource separately, you can see how Jobs, PodGroups, Queues, and Pods relate to one another.

This is especially useful when a workload is pending or not progressing as expected. The map can show the Job, its related PodGroup, the Pods created for the workload, and the Queue context around it. Warning and error states also make it easier to spot resources that need attention.

![Volcano resources in the Headlamp map view](volcano-map-view.png)

## Why use this alongside CLI tools

The plugin is not trying to replace `kubectl` or Volcano CLI. Those remain important for automation, scripting, and raw object inspection. What the plugin improves is the interactive troubleshooting experience: discovering related resources more quickly, understanding structured detail pages, and moving from scheduling state to runtime output without switching tools constantly.

## What’s next

> This work brings the main Volcano workflow into Headlamp, including Jobs, Queues, PodGroups, and the map view. Future work will build on this foundation with Prometheus integration, richer scheduling insights, and more workflow-oriented visibility across Volcano workloads.

Reviewer 4: "Future work will build…" makes a forward-looking promise; soften to "Possible future work includes…" so the post ages well if priorities shift.
